src/view/view.js

```diff
@@ -0,0 +1,374 @@
+	 */
+	scrollToTheSelection() {
+		const range = this.document.selection.getFirstRange();
+
+		if ( range ) {
+			scrollViewportToShowTarget( {
+				target: this.domConverter.viewRangeToDom( range ),
+				viewportOffset: 20
+			} );
+		}
+	}
+
+	/**
+	 * It will focus DOM element representing {@link module:engine/view/editableelement~EditableElement EditableElement}
+	 * that is currently having selection inside.
+	 */
+	focus() {
+		if ( !this.document.isFocused ) {
+			const editable = this.document.selection.editableElement;
+
+			if ( editable ) {
+				this.domConverter.focus( editable );
+				this.render();
+			} else {
+				/**
+				 * Before focusing view document, selection should be placed inside one of the view's editables.
+				 * Normally its selection will be converted from model document (which have default selection), but
+				 * when using view document on its own, we need to manually place selection before focusing it.
+				 *
+				 * @error view-focus-no-selection
+				 */
+				log.warn( 'view-focus-no-selection: There is no selection in any editable to focus.' );
+			}
+		}
+	}
+
+	/**
+	 * Change method is the primary way of changing the view. You should use it to modify any node in the view tree.
+	 * It makes sure that after all changes are made view is rendered to DOM. It prevents situations when DOM is updated
+	 * when view state is not yet correct. It allows to nest calls one inside another and still perform single rendering
+	 * after all changes are applied.
+	 *
+	 *		view.change( writer => {
+	 *			writer.insert( position1, writer.createText( 'foo' );
+	 *
+	 *			view.change( writer => {
+	 *				writer.insert( position2, writer.createText( 'bar' );
+	 *			} );
+	 *
+	 * 			writer.remove( range );
+	 *		} );
+	 *
+	 * Change block is executed immediately.
+	 *
+	 * When the outermost change block is done and rendering to DOM is over it fires
+	 * {@link module:engine/view/document~Document#event:change} event.
+	 *
+	 * @param {Function} callback Callback function which may modify the view.
+	 */
+	change( callback ) {
+		if ( this._renderingInProgress ) {
+			/**
+			 * Warning displayed when there is an attempt to make changes in the view tree during the rendering process.
+			 * This may cause unexpected behaviour and inconsistency between the DOM and the view.
+			 *
+			 * @error applying-view-changes-on-rendering
+			 */
+			log.warn(
+				'applying-view-changes-on-rendering: ' +
+				'Attempting to make changes in the view during rendering process. ' +
+				'This may cause some unexpected behaviour and inconsistency between the DOM and the view.'
+			);
+		}
+
+		// If other changes are in progress wait with rendering until every ongoing change is over.
+		if ( this._ongoingChange ) {
+			callback( this._writer );
+		} else {
+			this._ongoingChange = true;
+
+			callback( this._writer );
+			this._render();
+
+			this._ongoingChange = false;
+
+			this.document.fire( 'change' );
```

Comment:
As we talk, I am not sure about the order here. `change` event is fired after the rendering, what can be a suprose for someone. I think that rendering should be the result of the `change` event (listener with the `low` priority). Also both `change` and `render` should be inside `_ongoingChange` block, so there will be no infinite loop if one add some more change. Also if `view.change` blok is open after the rendering, in the `change` event, exception should be thrown.